### ISSUE TITLE
NavigationView integration

### ISSUE THREAD

**Issue by MaxDesiatov** [carson-katri/router#1]
I'm not sure how this would work yet, but it would be great if the router either supported the native `NavigationView` and `NavigationLink` types or introduced new `RoutableNavigation` and `RoutableLink` (names to bikeshed) types that behaved in a similar way with support for `NavigationView` layout and transitions. Maybe it is supported already? I haven't had a chance to explore the source code yet, feel free to close this then.

**Comment by carson-katri** [comment]
Yeah that'd be nice. It would probably need to use the `isActive` binding on `NavigationLink` to activate the routes.

**Comment by carson-katri** [comment]
Ok, so I've got it somewhat working with 1a896c5.

That added a new `routerLinkStyle` modifier and `RouterLinkStyle` type. There are `ButtonRouterLinkStyle`, `LinkButtonRouterLinkStyle`, and `NavigationLinkRouterLinkStyle`.
